### PR TITLE
[customcontext] Fix copying QImages from hybris textures. JB#58200

### PR DESCRIPTION
**Pull request by krnlyng**
When copying data out of a hybris texture take the buffer format into
account, otherwise we get rgb <-> bgr swapped images.

### CODE REVIEW

**Review by llewelld — APPROVED** [review]
Amazing work @krnlyng :+1: I tested it on an Xperia 10 (4.4.0.64) using rich text in a Label, Image components, using local, remote (from the Web), large and smaller images. I tested using code similar to Edz's MWE [from the forum](https://forum.sailfishos.org/t/qml-label-dispays-images-with-a-blue-ish-tint/8818/2).

Before the change, the remote rich text image had its channels swapped incorrectly, while all the other images were rendered correctly. After the change all of the images were rendered correctly. I didn't notice any other issues elsewhere in the OS (although I only checked this briefly).

Given all this, it seems to do the trick, so LGTM!

I'm approving now, but as you suggested earlier it'd be good to get feedback from the community before merging.

**Review by mkosola — APPROVED** [review]

**Comment by poetaster** [comment]
I just tested this with 2 cases. 
Label as in @llewelld first comment (but in the app Tidings). 
Canvas drawImage()

In both the color swapping is fixed. Thanks!

**Pull request merged**
